### PR TITLE
Atchub updatelogscollector

### PR DESCRIPTION
**Pull request by Atchub**
adding update Management files collection to the collector.
also testing to add update commands

version 2- adding worker.log and few update commands

### CODE REVIEW

**Comment by vrdmr** [comment]
@Atchub The travis CI is failing, but looks like not due to Patch management but something related to NPMD. @chparimi Could you please take a look?
```
================================================================================
Error: [40;33;1mtest_case_10_binary_replaced_by_x32(NPMDServerTest)[0m: IOError: stream closed
/home/travis/build/Microsoft/Build-OMS-Agent-for-Linux/omsagent/source/code/plugins/in_npmd_server.rb:251:in ``'
/home/travis/build/Microsoft/Build-OMS-Agent-for-Linux/omsagent/source/code/plugins/in_npmd_server.rb:251:in `kill_all_agent_instances'
/home/travis/build/Microsoft/Build-OMS-Agent-for-Linux/omsagent/source/code/plugins/in_npmd_server.rb:162:in `shutdown'
/home/travis/bin/omsagent_test_ruby/lib/ruby/gems/2.4.0/gems/fluentd-0.12.40/lib/fluent/test/base.rb:82:in `run'
/home/travis/bin/omsagent_test_ruby/lib/ruby/gems/2.4.0/gems/fluentd-0.12.40/lib/fluent/test/input_test.rb:119:in `run'
/home/travis/build/Microsoft/Build-OMS-Agent-for-Linux/omsagent/test/code/plugins/in_npmd_server_plugintest.rb:492:in `test_case_10_binary_replaced_by_x32'
     489: 
     490:         # Step 10
     491:         sleep(2)
[40;33;1m  => 492:         d.run[0m
     493: 
     494:         # Step 11
     495:         assert_equal(false, File.exist?("#{FAKE_BINARY_LOCATION}_x32"), "#{FAKE_BINARY_LOCATION}_x32 should have been deleted again")
================================================================================
```

File: tools/LogCollector/source/omslinux_agentlog.py [hunk 1]
```diff
@@ -456,6 +456,66 @@ def copyExtensionFiles():
...
+    return 0
+
+'''
+Return the package manager on the system
+'''
+
+def GetPackageManager():
+    ret = None
+    # choose default - almost surely one will match.
+    for b in ('apt-get', 'zypper', 'yum'):
```
Comment (DeveloperTommy) [inline tools/LogCollector/source/omslinux_agentlog.py]: Change b to more concrete variable

File: tools/LogCollector/source/omslinux_agentlog.py [hunk 2]
```diff
@@ -456,6 +456,66 @@ def copyExtensionFiles():
...
+'''
+Copy Update Management Solution logs into /tmp/updatelogs
+'''
+def copyUpdateFiles():
+    cmd='mkdir -p /tmp/omslogs/updateMgmtlogs'
+    out=execCommand(cmd)
+    writeLogCommand(cmd)
+
+    cmd='cp /var/opt/microsoft/omsagent/log/urp.log /tmp/omslogs/updateMgmtlogs'
+    out=execCommand2(cmd)
```
Comment (DeveloperTommy) [inline tools/LogCollector/source/omslinux_agentlog.py]: here you are using execCommand2, just want to be sure that this is the desired behavior.

File: tools/LogCollector/source/omslinux_agentlog.py [hunk 3]
```diff
@@ -456,6 +456,66 @@ def copyExtensionFiles():
     writeLogCommand(cmd)
     return 0
 
+'''
+Copy Update Management Solution logs into /tmp/updatelogs
```
Comment (DeveloperTommy) [inline tools/LogCollector/source/omslinux_agentlog.py]: replace /tmp/updatelogs with /tmp/omslogs/updateMgmtlogs

File: tools/LogCollector/source/omslinux_agentlog.py [hunk 4]
```diff
@@ -456,6 +456,66 @@ def copyExtensionFiles():
...
+    out=execCommand(cmd)
+    writeLogCommand(cmd)
+    return 0
+
+'''
+Return the package manager on the system
+'''
+
+def GetPackageManager():
+    ret = None
```
Comment (DeveloperTommy) [inline tools/LogCollector/source/omslinux_agentlog.py]: can we rename ret to something like pkgMgr

File: tools/LogCollector/source/omslinux_agentlog.py [hunk 5]
```diff
@@ -456,6 +456,66 @@ def copyExtensionFiles():
...
+    # choose default - almost surely one will match.
+    for b in ('apt-get', 'zypper', 'yum'):
+        code = execCommand2('which ' + b)
```
Comment (DeveloperTommy) [inline tools/LogCollector/source/omslinux_agentlog.py]: can we rename code to ret_code

**Comment by DeveloperTommy** [comment]
can you also merge master into this branch?

File: tools/LogCollector/source/omslinux_agentlog.py (hunk 2 above)
Comment (Atchub) [reply tools/LogCollector/source/omslinux_agentlog.py]: good eye!, fixed it ,did not impact as one returned the return code and other output for a cp command, so went unnoticed.

File: tools/LogCollector/source/omslinux_agentlog.py (hunk 1 above)
Comment (Atchub) [reply tools/LogCollector/source/omslinux_agentlog.py]: resolved

**Pull request merged**